File: frontend/internal-admin/src/api/order.ts
import request from '../utils/request'

export interface Order {
  id: string
  order_number: string
  customer_name: string
  customer_email?: string
  customer_phone?: string
  customer_address?: string
  total_amount?: number
  currency: string
  status: string
  priority: string
  ocr_confidence?: number
  items?: OrderItem[]
  created_at: string
  updated_at: string
}

export interface OrderItem {
  id: string
  order_id: string
  line_number: number
  mpn?: string
  manufacturer?: string
  description?: string
  quantity: number
  unit_price: number
  total_price: number
  ocr_confidence: number
  review_status: string
}

export interface OrderListResponse {
  total: number
  page: number
  page_size: number
  orders: Order[]
}

export interface OrderStats {
  total_orders: number
  pending_orders: number
  processing_orders: number
  approved_orders: number
  rejected_orders: number
  completed_orders: number
  today_orders: number
  week_orders: number
  month_orders: number
}

export function getOrders(params?: any): Promise<OrderListResponse> {
  return request({
    url: '/api/v1/orders',
    method: 'get',
    params
  })
}

export function getOrder(id: string): Promise<Order> {
  return request({
    url: `/api/v1/orders/${id}`,
    method: 'get'
  })
}

export function createOrder(data: Partial<Order>): Promise<{ id: string }> {
  return request({
    url: '/api/v1/orders',
    method: 'post',
    data
  })
}

export function updateOrder(id: string, data: Partial<Order>): Promise<Order> {
  return request({
    url: `/api/v1/orders/${id}`,
    method: 'put',
    data
  })
}

export function deleteOrder(id: string): Promise<void> {
  return request({
    url: `/api/v1/orders/${id}`,
    method: 'delete'
  })
}

export function reviewOrder(id: string, action: string, reviewNotes?: string): Promise<Order> {
  return request({
    url: `/api/v1/orders/${id}/review`,
    method: 'post',
    data: { action, review_notes: reviewNotes }
  })
}

export function uploadOrderFile(file: File, autoApproveThreshold?: number): Promise<{ order_id: string; status: string; ocr_confidence: number; requires_manual_review: boolean }> {
  const formData = new FormData()
  formData.append('file', file)
  if (autoApproveThreshold) {
    formData.append('auto_approve_threshold', autoApproveThreshold.toString())
  }

  return request({
    url: '/api/v1/orders/ocr',
    method: 'post',
    data: formData,
    headers: {
      'Content-Type': 'multipart/form-data'
    }
  })
}

export function getOrderStats(): Promise<OrderStats> {
  return request({
    url: '/api/v1/orders/stats/summary',
    method: 'get'
  })
}

export function batchApproveOrders(orderIds: string[]): Promise<{ success: boolean; approved_count: number; order_ids: string[] }> {
  return request({
    url: '/api/v1/orders/batch/approve',
    method: 'post',
    data: orderIds
  })
}

export function batchRejectOrders(orderIds: string[], reason: string): Promise<{ success: boolean; rejected_count: number; order_ids: string[]; reason: string }> {
  return request({
    url: '/api/v1/orders/batch/reject',
    method: 'post',
    data: { order_ids: orderIds, reason }
  })
}

export function exportOrders(format?: string, statusFilter?: string, startDate?: string, endDate?: string): Promise<{ download_url: string; expires_at: string }> {
  return request({
    url: '/api/v1/orders/export',
    method: 'get',
    params: { format, status_filter: statusFilter, start_date: startDate, end_date: endDate }
  })
}

export function getOrderTrend(days: number = 30): Promise<{ days: number; data: Array<{ date: string; count: number }> }> {
  return request({
    url: '/api/v1/orders/stats/trend',
    method: 'get',
    params: { days }
  })
}

// 别名用于 Dashboard
export { getOrders as getRecentOrders }

File: services/order-service/app/models/order.py
"""
订单数据模型
"""
from datetime import datetime
from enum import Enum
from typing import List, Optional

from pydantic import BaseModel, Field


class OrderStatus(str, Enum):
    """订单状态"""
    PENDING = "pending"              # 待处理
    PROCESSING = "processing"        # 处理中
    REVIEW_NEEDED = "review_needed"  # 需要审核
    APPROVED = "approved"            # 已审核通过
    REJECTED = "rejected"            # 已拒绝
    COMPLETED = "completed"          # 已完成
    CANCELLED = "cancelled"          # 已取消


class OrderPriority(str, Enum):
    """订单优先级"""
    LOW = "low"
    NORMAL = "normal"
    HIGH = "high"
    URGENT = "urgent"


class ReviewStatus(str, Enum):
    """行项目审核状态"""
    PENDING = "pending"           # 待审核
    VERIFIED = "verified"         # 已验证
    CORRECTED = "corrected"      # 已修正
    NEEDS_REVIEW = "needs_review" # 需要审核


class OrderItem(BaseModel):
    """订单行项目"""
    id: str
    order_id: str
    line_number: int
    mpn: Optional[str] = None
    manufacturer: Optional[str] = None
    description: Optional[str] = None
    quantity: int = 0
    unit_price: float = 0.0
    total_price: float = 0.0
    ocr_confidence: float = 0.0
    review_status: ReviewStatus = ReviewStatus.PENDING
    review_notes: Optional[str] = None
    matched_component_id: Optional[str] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)


class Order(BaseModel):
    """订单主表"""
    id: str
    order_number: str
    source_id: Optional[int] = None
    customer_name: Optional[str] = None
    customer_email: Optional[str] = None
    customer_phone: Optional[str] = None
    customer_address: Optional[str] = None
    total_amount: Optional[float] = None
    currency: str = "CNY"
    status: OrderStatus = OrderStatus.PENDING
    priority: OrderPriority = OrderPriority.NORMAL
    ocr_confidence: Optional[float] = None
    ocr_raw_text: Optional[str] = None
    ocr_structured_data: Optional[dict] = None
    review_notes: Optional[str] = None
    reviewed_by: Optional[str] = None
    reviewed_at: Optional[datetime] = None
    notes: Optional[str] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
    items: List[OrderItem] = Field(default_factory=list)


class OrderFilter(BaseModel):
    """订单过滤条件"""
    status: Optional[OrderStatus] = None
    priority: Optional[OrderPriority] = None
    customer_name: Optional[str] = None
    customer_email: Optional[str] = None
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    min_amount: Optional[float] = None
    max_amount: Optional[float] = None
    has_ocr_confidence_below: Optional[float] = None
